logisland-plugins/logisland-common-processors-plugin/src/main/java/com/hurence/logisland/processor/ModifyId.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2016 Hurence (bailet.thomas@gmail.com)
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hurence.logisland.processor;
+
+import com.google.common.collect.Lists;
+import com.hurence.logisland.annotation.documentation.CapabilityDescription;
+import com.hurence.logisland.annotation.documentation.Tags;
+import com.hurence.logisland.component.AllowableValue;
+import com.hurence.logisland.component.PropertyDescriptor;
+import com.hurence.logisland.record.Record;
+import com.hurence.logisland.validator.StandardValidators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+@Tags({"record", "id", "idempotent", "generate", "modify"})
+@CapabilityDescription("modify id of records or generate it following defined rules")
+public class ModifyId extends AbstractProcessor {
+
+    private static final long serialVersionUID = -270933070438408174L;
+
+    private static final Logger logger = LoggerFactory.getLogger(ModifyId.class);
+
+    public static final AllowableValue GENERATE_RANDOM_UUID = new AllowableValue("randomUuid", "generate a random uid",
+            "generate a randomUid using java library");
+
+    public static final AllowableValue GENERATE_HASH = new AllowableValue("hashFields", "generate a hash from fields",
+            "generate a hash from fields");
+
```

Comment:
add a third strategy (TYPE_TIME_HASH) based on the pattern

$record_type-$record_time-hash($fields.to.hash)